apps/api/src/app/integrations/usecases/create-integration/create-integration.usecase.ts

```diff
@@ -53,31 +68,29 @@ export class CreateIntegration {
         }),
       });
 
-      response = await this.integrationRepository.create({
+      const providerIdCapitalized = `${command.providerId.charAt(0).toUpperCase()}${command.providerId.slice(1)}`;
+      const defaultName =
+        providers.find((provider) => provider.id === command.providerId)?.displayName ?? providerIdCapitalized;
+      const name = command.name ?? defaultName;
+      const identifier = command.identifier ?? `${slugify(name, { lower: true, strict: true })}-${shortid.generate()}`;
+
+      const integrationEntity = await this.integrationRepository.create({
+        name,
+        identifier,
         _environmentId: command.environmentId,
         _organizationId: command.organizationId,
         providerId: command.providerId,
         channel: command.channel,
-        credentials: encryptCredentials(command.credentials),
+        credentials: encryptCredentials(command.credentials ?? {}),
         active: command.active,
       });
 
-      if (command.active && ![ChannelTypeEnum.CHAT, ChannelTypeEnum.PUSH].includes(command.channel)) {
-        await this.deactivateSimilarChannelIntegrations.execute({
-          environmentId: command.environmentId,
-          organizationId: command.organizationId,
-          integrationId: response._id,
-          channel: command.channel,
-          userId: command.userId,
-        });
-      }
```

Comment:
We need to remember to keep it depending on the feature flag, right? Or am I missing something?